Assets/Scripts/Saving/SaveDataManager.cs

```diff
@@ -0,0 +1,107 @@
+                {
+                    bf = new BinaryFormatter();
+                    SaveData initSaveData = new SaveData();
+                    initSaveData.currentSaveFileActive = fileNum;
+                    bf.Serialize(file, initSaveData);
+                    file.Close();
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error initializing file " + fileNum + "\n" + e.Message + "\n" + e.StackTrace);
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Saves variables in the SaveData class to a file.
+        /// </summary>
+        /// <param name="fileNum"> Which file slot to save to. </param>
+        public static void Save(int fileNum)
+        {
+            try
+            {
+                //using keyword ensures that file gets disposed of even if exceptions are thrown, ensuring that the file isn't left open.
+                using (file = File.Create(Application.persistentDataPath + "/saveData" + fileNum + ".dat"))
+                {
+                    bf = new BinaryFormatter();
+                    bf.Serialize(file, saveData);
+                    file.Close();
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error saving file " + fileNum + "\n" + e.Message + "\n" + e.StackTrace);
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Returns whether or not a save file exist at the specified file number.
+        /// </summary>
+        /// <param name="fileNum"> The file number for the save file we check for </param>
+        /// <returns></returns>
+        public static bool DoesSaveFileExist(int fileNum)
+        {
+            return File.Exists(Application.persistentDataPath + "/saveData" + fileNum + ".dat");
+        }
+
+        /// <summary>
+        /// Loads variables to the SaveData class from a file.
+        /// </summary>
+        /// <param name="fileNum"> Which file slot to load </param>
+        public static void Load(int fileNum)
+        {
+            //Create empty save file at this slot if one does not exist
+            if (!DoesSaveFileExist(fileNum))
+            {
+                Init(fileNum);
+            }
+            try
+            {
+                //using keyword ensures that file gets disposed of even if exceptions are thrown, ensuring that the file isn't left open.
+                using (file = File.Open(Application.persistentDataPath + "/saveData" + fileNum + ".dat", FileMode.Open))
+                {
+                    bf = new BinaryFormatter();
+                    saveData = (SaveData)bf.Deserialize(file);
+                    file.Close();
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error loading file " + fileNum + "\n" + e.Message + "\n" + e.StackTrace);
```

Comment:
We should actually have some kind of error handling here. Maybe the return type to a bool and leave it to whatever is calling this function?